posthog/migrations/0515_grouptypemapping_project_non_null.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 4.2.15 on 2024-10-15 13:32
+
+from django.db import migrations, models
+from django.contrib.postgres.operations import ValidateConstraint
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+    dependencies = [("posthog", "0514_errortrackingstackframe_context")]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                ValidateConstraint(
+                    model_name="grouptypemapping",
+                    name="group_type_project_id_is_not_null",
+                ),
+                # We can't use AddConstraintNotValid for UNIQUE CONSTRAINTs such as below, but these two here are
+                # actually fine, because UNIQUE CONSTRAINTs are backed by indexes - and we've already created the right
+                # indexes CONCURRENTLY in the preceding grouptypemapping_project_backfill migration.
+                # We signal safety here with `-- existing-table-constraint-ignore`
+                migrations.RunSQL(
+                    sql='ALTER TABLE "posthog_grouptypemapping" ADD CONSTRAINT "unique group types for project" UNIQUE ("project_id", "group_type"); -- existing-table-constraint-ignore',
```

Comment:
I'd generally like to prefer constraint names with no spaces in them, because it's just a little easier to specify in SQL (no need for quotation marks), but it's not a blocker.